contrib/cirrus/check_image.sh

```diff
@@ -36,4 +36,28 @@ do
         "$(systemctl list-unit-files --no-legend $REQ_UNIT)" = "$REQ_UNIT enabled" || let "RET+=1"
 done
 
+# Previously, perdiodic package updates were enabled and would trigger
+# soon after an Ubuntu VM booted, grabbing a specific lock file for
+# several minutes after setup_environment.sh ran (previous script here also).
+DPKGLCK=/var/lib/dpkg/lock
+if [[ "$OS_RELEASE_ID" == "ubuntu" ]] && [[ -r "$DPKGLCK" ]]
+then
+    item_test "The lock file $DPKGLCK exists" -r "$DPKGLCK" || let "RET+=1"
+    flock --exclusive --nonblock --no-fork $DPKGLCK sleep 1h &
+    sleep 1s  # just in case
+    FLOCKPID="$!"
+    item_test "Uncontested dpkg lock acquired" "$(jobs -r | wc -l)" -eq "1" || let "RET+=1"
+    echo "(Waiting 60 seconds for contestents to block)"
+    sleep 60s
+    # exit(1) means no processes found
+    CONTESTED=$(lsof -p ^$FLOCKPID -t -f -- $DPKGLCK &> /dev/null; echo $?)
```

Comment:
And here I get completely lost. Can you explain (in the header comments above) exactly what is going on here?

Reply:
This is to confirm anything that would grab the lock, has had a chance to block, and would show up in lsof.  But I see this is flawed, since it's only checking if another process has the lock file open, not actually locked (mandatory locks would likely be a PITA to turn on for /var).

Hmmm.  TBH, I have have a mind to just scrap trying to test this 'might happen'.  At best it's only going to give a false sense of security, e.g. if crond starts up and grabs the lock after 01:01.

Might be better to just confirm crond isn't running.

Reply:
code-deleted